### PR TITLE
Seperate lsb_dist detection from dist_version detection

### LINKED ISSUE #14924: Install script pulls LSB Codename as dist_version on RHEL/clones which isn't correct
In ```hack/install.sh```, the first method of determining distro name and version is via lsb_release:

    if command_exists lsb_release; then
                lsb_dist="$(lsb_release -si)"
                dist_version="$(lsb_release --codename | cut -f2)"
    fi

This works great for Debian/Ubuntu repos which use the codename in the path, but is not so great for RHEL/Fedora (and clones) who don't. If a RHEL/Fedora machine has lsb_release installed, the install.sh fails:

    # curl -sSL https://get.docker.com/ | sh
    sh: line 249: [: TwentyTwo: integer expression expected
    + sh -c 'sleep 3; yum -y -q install docker-engine'
    Yum command has been deprecated, redirecting to '/usr/bin/dnf -y -q install docker-engine'.
    See 'man dnf' and 'man yum2dnf' for more information.
    To transfer transaction metadata from yum to DNF, run:
    'dnf install python-dnf-plugins-extras-migrate && dnf-2 migrate'

    Failed to synchronize cache for repo 'docker-main-repo' from 'https://yum.dockerproject.org/repo/main/fedora/TwentyTwo': Cannot download repomd.xml: Cannot     download repodata/repomd.xml: All mirrors were tried, disabling.
    Error: no package matched: docker-engine

Because the ```dist_version``` returned from ```lsb_release --codename``` is ```TwentyTwo``` on Fedora. On RHEL, it'll return something like ```Maipo``` for RHEL7 and it'll return ```n/a``` on Oracle Linux (which doesn't use codenames).

### PR DESCRIPTION
**Pull request by Djelibeybi**
Fixes #14924. As discussed with @jfrazelle on IRC to make the lsb_dist/dist_version detection more flexible.

Signed-off-by: Avi Miller avi.miller@oracle.com

### CODE REVIEW

**Comment by Djelibeybi** [comment]
Ping @jfrazelle and @tianon for review.

**Comment by WPH95** [comment]
Separate `${lsb_dist}` and `${dist_version}` is indeed a better solution!

**Comment by yank1** [comment]
+1

**Comment by jfrazelle** [comment]
ping @erikh does this fix your issue?

**Comment by erikh** [comment]
```
[root@ceph-mon0 yum.repos.d]# curl https://raw.githubusercontent.com/Djelibeybi/docker/fix14924/hack/install.sh | sudo bash
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  9752  100  9752    0     0  25300      0 --:--:-- --:--:-- --:--:-- 25329
+ sh -c 'sleep 3; yum -y -q install docker-engine'
warning: /var/cache/yum/x86_64/7/docker-main-repo/packages/docker-engine-1.7.1-1.el7.centos.x86_64.rpm: Header V4 RSA/SHA1 Signature, key ID 2c52609d: NOKEY
Public key for docker-engine-1.7.1-1.el7.centos.x86_64.rpm is not installed
Importing GPG key 0x2C52609D:
 Userid     : "Docker Release Tool (releasedocker) <docker@docker.com>"
 Fingerprint: 5811 8e89 f3a9 1289 7c07 0adb f762 2157 2c52 609d
 From       : https://yum.dockerproject.org/gpg

If you would like to use Docker as a non-root user, you should now consider
adding your user to the "docker" group with something like:

  sudo usermod -aG docker your-user

Remember that you will have to log out and back in for this to take effect!
```

I’d say yes :) Thanks so much @djelibeybi and @jessfraz!

**Comment by erikh** [comment]
Actually no, this doesn't necessarily fix experimental. Let me hack up the script a touch to fix the hostname and validate again.

**Comment by Djelibeybi** [comment]
@erikh I didn't touch the hostname bits, so that's identical to upstream. This should just fix the lsb_dist and dist_version determination.

**Comment by erikh** [comment]
I have validated this works for both experimental and release on centos 7.

**Comment by calavera** [comment]
@Djelibeybi it looks like we need to rebase your branch. Can you take care of it?

**Comment by Djelibeybi** [comment]
Rebase committed.

**Comment by jfrazelle** [comment]
ok let me run my test script on this, also ping @tianon

**Comment by jfrazelle** [comment]
LGTM

**Comment by runcom** [comment]
This can be now removed https://github.com/docker/docker/blob/master/hack/install.sh#L54

File: hack/install.sh [hunk 1]
```diff
@@ -263,8 +285,8 @@ do_install() {
 			exit 0
 			;;
 
-		fedora|centos)
-			$sh_c "cat >/etc/yum.repos.d/docker-${repo}.repo" <<-EOF
```
Comment (calavera) [inline hack/install.sh]: @jfrazelle, @Djelibeybi was this line supposed to go away? didn't we add it a few days ago?

Comment (jfrazelle) [inline hack/install.sh]: i made a pr i pinged u

**Comment by calavera** [comment]
cherry picked into #15091

**Pull request merged**